### PR TITLE
Performance improvements to the name equality check when finding a repository index.

### PR DESCRIPTION
**Pull request by jonathan-aotearoa**
#### Check List:

- [x] You have run `./mvnw verify` and the project builds successfully
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
- [x] For new entries, or after substantial changes: When implementing custom hash structures, please point to where you deal with hash collisions (line number)

* Execution time:
* Execution time of reference implementation:

### CODE REVIEW

**Comment by jonathan-aotearoa** [comment]
See line 544 for relevant changes

File: src/main/java/dev/morling/onebrc/CalculateAverage_jonathanaotearoa.java [hunk 1]
```diff
@@ -510,46 +503,65 @@ private int findIndex(int nameHash, final long nameAddress, final byte nameSize)
...
 
         /**
-         * Checks if two locations in memory have the same value.
+         * Checks if the names at the specified locations meet our threshold for equality.
+         * <p>
+         * It's possible, due to hash collisions, that two different names have the same size and hash.
+         * To account for this, we also need to check if the station names themselves are equal.
+         * However, checking all the bytes in both names is costly.
+         * We therefore set a threshold for the maximum number of bytes, at the start and end of the name, to check.
+         * If two names with the same size and hash have the same first N and last N bytes, we're happy they are the equal.
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_jonathanaotearoa.java:536]: I'm afraid that this is not valid, as it isn't guaranteed to resolve all potential conflicts.

**Comment by gunnarmorling** [comment]
A tad slower actually than before:

Benchmark 1: timeout -v 300 ./calculate_average_jonathan-aotearoa.sh 2>&1
  Time (mean ± σ):      5.103 s ±  0.015 s    [User: 32.934 s, System: 0.715 s]
  Range (min … max):    5.091 s …  5.128 s    5 runs

Summary
  jonathan-aotearoa: trimmed mean 5.098993353046667, raw times 5.09063945838,5.09602143738,5.10543017038,5.09552845138,5.128256689380001

Leaderboard
grep: ./src/main/java*/dev/morling/onebrc/CalculateAverage_jonathan-aotearoa.java: No such file or directory

| # | Result (m:s.ms) | Implementation     | JDK | Submitter     | Notes     |
|---|-----------------|--------------------|-----|---------------|-----------|
|   | 00:05.098 | [link](https://github.com/gunnarmorling/1brc/blob/main/src/main/java/dev/morling/onebrc/CalculateAverage_jonathan-aotearoa.java)| 21.0.2-graal | [Jonathan Wright](https://github.com/jonathan-aotearoa) | GraalVM native binary |

**Comment by jonathan-aotearoa** [comment]
I did some benchmarking on that method and the results weren't conclusive, so I'm not entirely surprised to see it ran fractionally slower. The issue I'm finding with micro-benchmarking methods in isolation is that the results don't always apply when the method is run in the context of the whole application. It's a great learning experience though. I haven't had an excuse to dive into Linux tools like `perf` before :)

**Comment by gunnarmorling** [comment]
Nice to hear :) Wanna close this one then?

**Comment by jonathan-aotearoa** [comment]
> Hey @jonathan-aotearoa, I am gonna close this one, as we're after the cut-off time, and this one didn't yield an improvement. Thanks a lot for participating in 1BRC!

Hi @gunnarmorling, thanks for closing the issue, and thanks again for taking the time to setup and and administer this challenge. Very much appreciated.

**Pull request closed without merging**